src/sync.d

```diff
@@ -442,6 +446,10 @@ final class SyncEngine
 		if (isSymlink(path) && !exists(readLink(path))) {
 			return;
 		}
+		if (!isValidName(path)) {
```

Comment:
This logic is duplicated, can this be split out into a function or otherwise moved to a single location?